http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java

```diff
@@ -0,0 +1,457 @@
+// SPDX-License-Identifier: Apache 2.0
+
+package org.fido.iot.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.Calendar;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+import org.fido.iot.protocol.Composite;
+import org.fido.iot.protocol.Const;
+import org.fido.iot.serviceinfo.ModuleManager;
+
+public class OwnerSystemResourceServlet extends HttpServlet {
+
+  public static final String BYTES_TAG = "bytes";
+  public static final String ID_TAG = "id";
+  public static final String CRID_TAG = "crid";
+  public static final String GUID_TAG = "guid";
+  public static final String DEVICE_TAG = "device";
+  public static final String OS_TAG = "os";
+  public static final String VERSION_TAG = "version";
+  public static final String ARCH_TAG = "arch";
+  public static final String FILENAME_TAG = "filename";
+  public static final String PRIORITY_TAG = "priority";
+  public static final String HASH_TAG = "hash";
+  public static final String MODULE_TAG = "module";
+  public static final String VAR_TAG = "var";
+  public static final String WHERE_AND = " AND ";
+
+
+  private byte[] getQueryBytes(String queryArg) {
+    StringBuilder hex = new StringBuilder();
+    for (char c : queryArg.toCharArray()) {
+      if (Character.isLetterOrDigit(c)) {
+        hex.append(c);
+      }
+    }
+
+    return Composite.decodeHex(hex.toString());
+  }
+
+  private String getWhereClause(HttpServletRequest req) {
+    String id = req.getParameter(ID_TAG);
+    String crid = req.getParameter(CRID_TAG);
+    String guid = req.getParameter(GUID_TAG);
+    String device = req.getParameter(DEVICE_TAG);
+    String os = req.getParameter(OS_TAG);
+    String version = req.getParameter(VERSION_TAG);
+    String arch = req.getParameter(ARCH_TAG);
+    String fileName = req.getParameter(FILENAME_TAG);
+    String priority = req.getParameter(PRIORITY_TAG);
+    String hash = req.getParameter(HASH_TAG);
+    String module = req.getParameter(MODULE_TAG);
+    String varName = req.getParameter(VAR_TAG);
+    String contentType = null;
+
+    if (module != null && varName != null) {
+      contentType = module + ModuleManager.MODULE_DELIMITER + varName;
+    }
+
+    StringBuilder whereString = new StringBuilder("WHERE ");
+
+    if (id != null) {
+      whereString.append("RESOURCE_ID = ");
+      whereString.append(id.toString());
+    } else {
+      String and = "";
+      if (crid != null) {
+        whereString.append(and);
+        whereString.append("CONTENT_RESOURCE_TAG = ");
+        whereString.append(crid.toString());
+        and = WHERE_AND;
```

Comment:
An easier why to handle the "AND" clause is to structure the  where clause to start with "where 1=1". Then each subsequent optional item can always include the "and". For example: "and some_id = value"

Reply:
The reason I did it this way is that I'm not certain  which column will be first.  The first non-null query argument will be the first  in the where clause but any of the columns can to the first.  Whatever strategy I use I will need to keep track using some flag for state on whether the its the first column or non-first column.

Reply:
That is why the where clause always starts with "where 1=1". Then each subsequent clause will always be of the form "and foo=bar". In this case: "and ZZZ_TAG = value".